server/src/main/java/com/broadcom/lsp/cobol/core/visitor/VariableDefinitionDelegate.java

```diff
@@ -253,75 +287,59 @@ private String retrieveQualifier(String name) {
-            .orElseGet(
-                () -> {
-                  addError(
-                      messages.getMessage(CANNOT_BE_RENAMED, it.getName()), it.getDefinition());
-                  return null;
-                });
-  }
-
   @Nullable
-  private ElementItem getConditionalContainer() {
-    return (ElementItem)
-        ofNullable(variables.peek())
-            .filter(it -> it instanceof ElementItem)
-            .filter(it -> ((ElementItem) it).getValue().isEmpty())
-            .orElse(null);
-  }
-
-  @NonNull
-  private Locality retrieveDefinition(EntryNameContext context) {
-    return positions.get(context.start);
+  private Variable getConditionalContainer() {
+    return ofNullable(variables.peek())
+        .filter(Conditional::isConditional)
+        .orElseGet(
+            () ->
+                ofNullable(variables.peek())
+                    .filter(it -> it instanceof ConditionDataName)
+                    .map(Variable::getParent)
+                    .orElse(null));
   }
 
   @NonNull
-  private String retrievePicText(@NonNull DataPictureClauseContext picClause) {
-    return picClause.getText().replaceAll(picClause.getStart().getText(), "");
+  private Locality retrieveDefinition(ParserRuleContext context) {
+    return ofNullable(context).map(ParserRuleContext::getStart).map(positions::get).orElse(null);
   }
 
   @NonNull
-  private <T extends ParserRuleContext> Locality retrieveRangeLocality(@NonNull List<T> clauses) {
-    Locality start = positions.get(clauses.get(0).getStart());
-    Locality end = positions.get(clauses.get(clauses.size() - 1).getStop());
-    Range range = new Range(start.getRange().getStart(), end.getRange().getEnd());
-    return Locality.builder().uri(start.getUri()).range(range).build();
+  private String retrievePicText(List<DataPictureClauseContext> clauses) {
+    if (clauses.isEmpty()) return "";
+    DataPictureClauseContext clause = clauses.get(0);
+    return clause.getText().replaceAll(clause.getStart().getText(), "").trim();
   }
 
   private int retrieveOccursTimes(@NonNull DataOccursClauseContext dataOccursClauseContexts) {
-    return Integer.parseInt(dataOccursClauseContexts.integerLiteral().toString());
+    return Integer.parseInt(dataOccursClauseContexts.integerLiteral().getText());
   }
 
   @NonNull
   private List<IndexItem> retrieveIndexItem(@NonNull DataOccursClauseContext clause) {
     return clause.indexName().stream()
-        .map(it -> new IndexItem(it.getText(), it.getText(), positions.get(it.start)))
+        .map(
+            it ->
+                new IndexItem(
+                    it.getText().toUpperCase(),
+                    VariableUtils.createQualifier(it.getText().toUpperCase()),
+                    positions.get(it.start)))
         .collect(toList());
   }
 
+  private UsageFormat retrieveUsageFormat(List<DataUsageClauseContext> clauses) {
+    if (clauses.isEmpty()) return UsageFormat.UNDEFINED;
```

Comment:
why no null check?   clauses !=null && clauses.isEmpty() ?

Reply:
The grammar guarantees the clauses to be non-null